src/hooks/usePaginatedActions.ts

```diff
@@ -37,28 +37,35 @@ export const usePaginatedActions = (): UsePaginatedActionsReturn => {
     fetchPolicy: 'network-only',
   });
   const { items, nextToken } = data?.getActionsByColony || {};
+  const [visibleActions, setVisibleActions] = useState(ITEMS_PER_PAGE);
+
   const actions = items?.filter(notNull) || [];
-  const hasMoreActions = !!nextToken;
+  const hasMoreActions = visibleActions < actions.length || !!nextToken;
 
   const handleSortDirectionChange: SortDirectionChangeHandler = (
     newSortDirection,
   ) => {
     setSortDirection(newSortDirection);
   };
 
-  const loadMoreActions = () => {
+  const fetchMoreActions = nextToken && actions.length < visibleActions;
+  if (fetchMoreActions) {
     fetchMore({
       variables: {
         nextToken,
       },
     });
+  }
+
+  const loadMoreActions = () => {
+    setVisibleActions((count) => count + ITEMS_PER_PAGE);
```

Comment:
Every time the user requests more actions, we increase the number of "visibleActions". 

The component will rerender and if it needs to fetch more actions from the db, it will (line 52 above).